tests/ui/async-await/async-closures/once.rs

```diff
@@ -1,18 +1,20 @@
 //@ aux-build:block-on.rs
 //@ edition:2021
 //@ build-pass
+//@ revisions: current next
+//@[next] compile-flags: -Znext-solver
 
 #![feature(async_closure)]
 
 use std::future::Future;
 
 extern crate block_on;
 
-struct NoCopy;
+// Check that async closures always implement `FnOnce`
 
 fn main() {
     block_on::block_on(async {
-        async fn call_once<F: Future>(x: impl Fn(&'static str) -> F) -> F::Output {
+        async fn call_once<F: Future>(x: impl FnOnce(&'static str) -> F) -> F::Output {
```

Comment:
Even though this test was called `once.rs`, it was actually testing this behavior. Let's... fix that... by copying this into a properly named test, and changing this to actually test `FnOnce`.